distribution/openrct2.d.ts

```diff
@@ -1586,10 +1602,80 @@ declare global {
         removeElement(index: number): void;
     }
 
+    type ObjectSourceGame =
+        "rct1" |
+        "added_attractions" |
+        "loopy_landscapes" |
+        "rct2" |
+        "wacky_worlds" |
+        "time_twister" |
+        "custom" |
+        "openrct2_official";
+
+    type ObjectGeneration = "dat" | "json";
+
+    /**
+     * Represents an installed OpenRCT2 object which may or may not be currently loaded into the park.
+     */
+    interface InstalledObject {
+        /**
+         * The full path of the object file.
+         */
+        readonly path: string;
```

Comment:
Could this leak user information, like the user profile name on Windows?

Reply:
Yes, although it could only communicate externally via multiplayer.

Reply:
Or via the socket API. Would it be an option to trim it to a local path?

E.g. `C:/Users/<name>/Documents/OpenRCT2/object/customobj.parkobj` will turn into `OpenRCT2/object/customobj.parkobj`?

Reply:
The socket API can only communicate with localhost by default.

What if a plugin wants to show the full path much like the built in windows already do when debug mode is switched on?

Reply:
Objects shipped with OpenRCT2 are in the `/data/` folder while custom objects are not, so it's still recognisable by path where it comes from.

Personally I'd prefer a privacy first focus by not taking the risk to expose personal information, but I'm not sure what others think of it.

Reply:
The `OpenRCT2` portion of the truncated path you suggested is still whatever the user likes it to be. Why is a path considered sensitive information? It is already exposed via dump files which appear on the GitHub issues (for each one you can see the full path to where the crash dump was saved).

Reply:
The path is not necessarily sensitive, but it can contain identifiable information like someone's full name on Windows. And it's possible that exposing someone's full name without a user's explicit consent is a violation of laws like EU's GDPR. Same may apply to exposing someone's full name in the dump files.

But then again I'm far from a lawyer.

Reply:
So whatever the path is, I don't care. I'm just sharing a worry about the exposing identifiable information part.